test/integration/vorpal-engine_test.go

```diff
@@ -34,26 +40,102 @@ func TestScanVorpal_SentFileWithoutExtension_FailCommandWithError(t *testing.T)
 	assert.ErrorContains(t, err, errorConstants.FileExtensionIsRequired)
 }
 
-func TestExecuteVorpalScan_VorpalLatestVersionSetTrue_SuccessfullyReturnMockData(t *testing.T) {
+func TestExecuteVorpalScan_VorpalLatestVersionSetTrue_Success(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "",
+		flag(commonParams.VorpalLatestVersion),
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
 
-	scanResult, _ := commands.ExecuteVorpalScan("data/python-vul-file.py", true)
-	expectedMockResult := commands.ReturnSuccessfulResponseMock()
-	//TODO: update mocks when there's a real engine
-	assert.DeepEqual(t, scanResult, expectedMockResult)
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, "Sending empty source file should not fail")
 }
 
-func TestExecuteVorpalScan_VorpalLatestVersionSetFalse_SuccessfullyReturnMockData(t *testing.T) {
+func TestExecuteVorpalScan_VorpalLatestVersionSetFalse_Success(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/python-vul-file.py",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
 
-	scanResult, _ := commands.ExecuteVorpalScan("data/python-vul-file.py", false)
-	expectedMockResult := commands.ReturnFailureResponseMock()
-	//TODO: update mocks when there's a real engine
-	assert.DeepEqual(t, scanResult, expectedMockResult)
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, fmt.Sprintf("Should not fail with error: %v", err))
 }
 
 func TestExecuteVorpalScan_CorrectFlagsSent_SuccessfullyReturnMockData(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/python-vul-file.py",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
+
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, fmt.Sprintf("Should not fail with error: %v", err))
+}
+
+func TestExecuteVorpalScan_UnsupportedLanguage_Fail(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/positive1.tf",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
+
+	err, bytes := executeCommand(t, args...)
+	assert.NilError(t, err, "Scan should not fail with error")
+	var scanResult grpcs.ScanResult
+	err = json.Unmarshal(bytes.Bytes(), &scanResult)
+	assert.NilError(t, err, "Failed to unmarshal scan result")
+	assert2.NotNil(t, scanResult.Error)
+}
+
+func TestExecuteVorpalScan_InitializeAndRunUpdateVersion_Success(t *testing.T) {
+	configuration.LoadConfiguration()
```

Comment:
what is this test? 
if you want to check the starting of the engine it is enough to stop on line 108
and maybe you can check that the port changed

Reply:
I thought about checking if the port changed but after thinking about it, there is a scenario that getPort will return the same port because it is not in use.

Reply:
this test can fail if the Vorpal is not installed yet

Reply:
It's ok. I'm running first : args := []string{
		"scan", "vorpal",
		flag(commonParams.SourcesFlag), "",
		flag(commonParams.VorpalLatestVersion),
		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
	}
	which install the vorpal engine if not exist